src/planner.c

```diff
@@ -737,6 +781,124 @@ contain_param(Node *node)
 }
 
+static List *
+fill_missing_groupclause(List *new_groupclause, List *orig_groupclause)
+{
+	if (new_groupclause != NIL)
+	{
+		ListCell *gl;
+		foreach (gl, orig_groupclause)
+		{
+			SortGroupClause *gc = lfirst_node(SortGroupClause, gl);
+
+			if (list_member_ptr(new_groupclause, gc))
+				continue; /* already in list */
+			new_groupclause = lappend(new_groupclause, gc);
+		}
+	}
+	return new_groupclause;
+}
+
+static bool
+check_cagg_view_rte(RangeTblEntry *rte)
+{
+	ContinuousAgg *cagg = NULL;
+	ListCell *rtlc;
+	bool found = false;
+	Query *viewq = rte->subquery;
+	Assert(rte->rtekind == RTE_SUBQUERY);
+
+	if (list_length(viewq->rtable) != 3) /* a view has 3 entries */
+	{
+		return false;
+	}
+
+	// should cache this information for cont. aggregates
+	foreach (rtlc, viewq->rtable)
+	{
+		RangeTblEntry *rte = (RangeTblEntry *) lfirst(rtlc);
+		char *schema;
+		char *table;
+
+		if (!OidIsValid(rte->relid))
+			break;
+
+		schema = get_namespace_name(get_rel_namespace(rte->relid));
+		table = get_rel_name(rte->relid);
+		if ((cagg = ts_continuous_agg_find_by_view_name(schema, table)) != NULL)
+			found = true;
+	}
+	return found;
+}
+
+/* Note that it modifies the passed in Query
+* select * from (select a, b, max(c), min(d) from ...
+				 group by a, b)
+  order by b;
+* is transformed as
+* SELECT * from (select a, b, max(c), min(d) from ..
+*                 group by B, A  <------ note the change in order here
+*              )
+*  order by b;
+*  we pushdown only if order by is a subset of group-by
+* restrict this to continuous aggregates
+* Parameters:
+* subq_rte - rte for subquery (inner query that will be modified)
+* outer_sortcl -- outer query's sort clause
+* outer_tlist - outer query's target list
+*/
+static void
+cagg_reorder_groupby_clause(RangeTblEntry *subq_rte, int rtno, List *outer_sortcl,
+							List *outer_tlist)
+{
+	bool not_found = true;
+	Query *subq;
+	ListCell *lc;
+	Assert(subq_rte->rtekind == RTE_SUBQUERY);
+	subq = subq_rte->subquery;
+	if (outer_sortcl && subq->groupClause && subq->sortClause == NIL &&
+		check_cagg_view_rte(subq_rte))
+	{
+		List *new_groupclause = NIL;
+		/* we are going to modify this. so make a copy and use it
+		 if we replace */
+		List *subq_groupclause_copy = copyObject(subq->groupClause);
+		foreach (lc, outer_sortcl)
+		{
+			SortGroupClause *outer_sc = (SortGroupClause *) lfirst(lc);
+			TargetEntry *outer_tle = get_sortgroupclause_tle(outer_sc, outer_tlist);
+			not_found = true;
+			if (IsA(outer_tle->expr, Var) && (((Var *) outer_tle->expr)->varno == rtno))
+			{
+				int outer_attno = ((Var *) outer_tle->expr)->varattno;
+				TargetEntry *subq_tle = list_nth(subq->targetList, outer_attno - 1);
+				if (subq_tle->ressortgroupref > 0)
+				{
+					// get group clause corresponding to this.
+					SortGroupClause *subq_gclause =
+						get_sortgroupref_clause(subq_tle->ressortgroupref, subq_groupclause_copy);
+					if (subq_gclause->sortop != outer_sc->sortop)
```

Comment:
again, I see no reason why the sortops and nulls_first are aligned, both need to be fixed up. You can do ORDER BY X DESC NULLS FIRST or ORDER BY X DESC NULLS LAST

Reply:
modified and added test case for this.